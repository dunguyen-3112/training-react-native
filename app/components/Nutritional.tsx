import { FlatList, SectionList, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { NutritionalProps } from './Food';
import { COLOR } from '@constants';
import CustomText from './CustomText';

const Nutritional = ({
  nutritional,
  color,
  marginTop = 0,
}: {
  nutritional: NutritionalProps;
  color: COLOR;
  marginTop?: number;
}) => {
  const vi = Object.entries(nutritional).map(([x, y]) => ({
    key1: x,
    value: y,
  }));

  console.log(vi);

  return (
    <View style={[styles.container, { marginTop }]}>
      <FlatList
        data={vi}
        horizontal
        renderItem={({ item }) => <NutritionalItem {...item} />}
        ItemSeparatorComponent={() => <View style={{ marginLeft: 40 }} />}
        keyExtractor={(item) => item.key1}
      />
    </View>
  );
};

const NutritionalItem = ({ key1, value }: { key1: string; value: number }) => {
  return (
    <View>
      <CustomText text={key1} size={16} />
      <CustomText text={`${value}g`} size={24} color={'orange'} />
    </View>
  );
};

export default Nutritional;

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFF8EE',
    justifyContent: 'center',
    paddingVertical: 19,
    alignItems: 'center',
  },
});
