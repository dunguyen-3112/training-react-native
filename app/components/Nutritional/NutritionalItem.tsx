import Text from '@components/common/Text';
import React from 'react';
import { View } from 'react-native';

const NutritionalItem = ({ key1, value }: { key1: string; value: number }) => {
  return (
    <View>
      <Text font={{ fontSize: 16, textTransform: 'capitalize' }}>{key1}</Text>
      <Text font={{ fontSize: 24 }} color={'orange'}>
        {`${value}g`}
      </Text>
    </View>
  );
};

export default NutritionalItem;
