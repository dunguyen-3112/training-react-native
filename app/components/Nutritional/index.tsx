import { FlatList, StyleSheet, View, ViewStyle } from 'react-native';
import React, { memo } from 'react';
import { COLOR } from '@constants';
import { TNutritional } from '@types';
import NutritionalItem from './NutritionalItem';

const Nutritional = ({
  nutritional,
  customStyles,
}: {
  nutritional: TNutritional;
  color?: COLOR;
  customStyles?: ViewStyle;
}) => {
  const vi = Object.entries(nutritional).map(([x, y]) => ({
    key1: x,
    value: y,
  }));

  return (
    <View style={[styles.container, customStyles]}>
      <FlatList
        data={vi}
        horizontal
        renderItem={({ item }) => <NutritionalItem {...item} />}
        ItemSeparatorComponent={() => <View style={{ marginLeft: 40 }} />}
        keyExtractor={(item) => item.key1}
      />
    </View>
  );
};

export default memo(Nutritional);

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFF8EE',
    justifyContent: 'center',
    paddingVertical: 19,
    alignItems: 'center',
  },
});
