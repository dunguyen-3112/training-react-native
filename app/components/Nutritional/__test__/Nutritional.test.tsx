import { TNutritional } from '@types';
import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';

import Nutritional from '../index';

describe('NutritionalItem', () => {
  const nutritional: TNutritional = {
    carbs: 200,
    fat: 200,
    protein: 200,
    calories: 200,
  };

  let component: ReactTestRenderer;
  beforeEach(() => {
    component = renderer.create(<Nutritional nutritional={nutritional} />);
  });

  it('renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
