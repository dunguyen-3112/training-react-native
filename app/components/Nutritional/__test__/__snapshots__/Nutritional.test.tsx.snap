// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NutritionalItem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF8EE",
        "justifyContent": "center",
        "paddingVertical": 19,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "title": "carbs",
          "value": 200,
        },
        Object {
          "title": "fat",
          "value": 200,
        },
        Object {
          "title": "protein",
          "value": 200,
        },
        Object {
          "title": "calories",
          "value": 200,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#5C5C5C",
                  "color": "#5C5C5C",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "textTransform": "capitalize",
                },
                undefined,
                Object {},
              ]
            }
          >
            carbs
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#FF8473",
                  "color": "#FF8473",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "textTransform": "none",
                },
                undefined,
                Object {},
              ]
            }
          >
            200g
          </Text>
        </View>
        <View
          style={
            Object {
              "marginLeft": 40,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#5C5C5C",
                  "color": "#5C5C5C",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "textTransform": "capitalize",
                },
                undefined,
                Object {},
              ]
            }
          >
            fat
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#FF8473",
                  "color": "#FF8473",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "textTransform": "none",
                },
                undefined,
                Object {},
              ]
            }
          >
            200g
          </Text>
        </View>
        <View
          style={
            Object {
              "marginLeft": 40,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#5C5C5C",
                  "color": "#5C5C5C",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "textTransform": "capitalize",
                },
                undefined,
                Object {},
              ]
            }
          >
            protein
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#FF8473",
                  "color": "#FF8473",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "textTransform": "none",
                },
                undefined,
                Object {},
              ]
            }
          >
            200g
          </Text>
        </View>
        <View
          style={
            Object {
              "marginLeft": 40,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#5C5C5C",
                  "color": "#5C5C5C",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "textTransform": "capitalize",
                },
                undefined,
                Object {},
              ]
            }
          >
            calories
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "borderColor": "#FF8473",
                  "color": "#FF8473",
                  "fontSize": 24,
                  "fontWeight": "400",
                  "textTransform": "none",
                },
                undefined,
                Object {},
              ]
            }
          >
            200g
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
